splinepy/helpme/fit.py

```diff
@@ -0,0 +1,738 @@
+            points[1:-1, :] += (
+                -coefficient_matrix[1:-1, [0]] * points[0, :]
+                - coefficient_matrix[1:-1, [-1]] * points[-1, :]
+            ).todense()
+
+            # solve system A^TAx=A^Tb
+            at = coefficient_matrix[1:-1, 1:-1].transpose(copy=True)
+            target_spline.control_points[1:-1] = _spsolve(
+                at @ coefficient_matrix[1:-1, 1:-1], at @ points[1:-1]
+            )
+
+            # alternative using lsqr
+
+            # for i in range(0, points.shape[1]):
+            #     (
+            #         target_spline.control_points[1:-1, i]
+            #     ) = _lsqr(coefficient_matrix[1:-1, 1:-1], points[1:-1, i])[0]
+
+            residual = _np.linalg.norm(
+                coefficient_matrix @ target_spline.control_points - points
+            )
+        else:
+            # move known values to RHS
+            points[1:-1, :] += (
+                -coefficient_matrix[1:-1, [0]] * points[0, :]
+                - coefficient_matrix[1:-1, [-1]] * points[-1, :]
+            )
+
+            # solve system using lstsq
+
+            (
+                target_spline.control_points[1:-1],
+                residual,
+            ) = _np.linalg.lstsq(
+                coefficient_matrix[1:-1, 1:-1], points[1:-1]
+            )[:2]
+            residual = _np.sqrt(_np.linalg.norm(residual))
+
+            # manual alternative: A^TAx=A^Tb
+
+            # at=coefficient_matrix[1:-1, 1:-1].copy().transpose()
+            # target_spline.control_points[1:-1]=_np.linalg.solve(
+            #     at@coefficient_matrix[1:-1, 1:-1],at@points[1:-1]
+            #     )
+            # residual = _np.linalg.norm(
+            #     coefficient_matrix @ target_spline.control_points - points
+            #     )
+
+    # interpolation or approximation with original system
+    elif _has_scipy:
+        at = coefficient_matrix.transpose(copy=True)
+        target_spline.control_points = _spsolve(
+            at @ coefficient_matrix, at @ points
+        )
+
+        # alternative using lsqr
+
+        # for i in range(0, points.shape[1]):
+        #     target_spline.control_points[:, i], _, _, residual[i] = _lsqr(
+        #         coefficient_matrix, points[:, i]
+        #     )[:4]
+
+        residual = _np.linalg.norm(
+            coefficient_matrix @ target_spline.control_points - points
+        )
+
+    else:
+        target_spline.control_points, residual = _np.linalg.lstsq(
+            coefficient_matrix, points
+        )[:2]
+        residual = _np.sqrt(_np.linalg.norm(residual))
+
+        # manual alternative
+
+        # at=coefficient_matrix.copy().transpose()
+        # target_spline.control_points=_np.linalg.solve(
+        #     at@coefficient_matrix,at@points
+        #     )
+        # residual = _np.linalg.norm(
+        #     coefficient_matrix @ target_spline.control_points - points
+        #     )
+
+    return residual
+
+
+def fit_curve(
```

Comment:
```suggestion
def curve(
```

The function will then be called using `splinpy.helpme.fit.curve`